test/select_io_dset.c

```diff
@@ -106,6 +106,21 @@ typedef enum {
 #define TEST_TCONV_BUF_TOO_SMALL               0x100
 #define TEST_IN_PLACE_TCONV                    0x200
 
+static herr_t
+check_actual_selection_io_mode(hid_t dxpl, uint32_t sel_io_mode_expected)
+{
+    uint32_t actual_sel_io_mode;
+
+    if (H5Pget_actual_selection_io_mode(dxpl, &actual_sel_io_mode) < 0)
+        FAIL_STACK_ERROR;
```

Comment:
API calls should probably use TEST_ERROR, not FAIL_STACK_ERROR, which will duplicate the error stack dump (TEST_ERROR already dumps the stack when h5_reset() has been called)